client/src/components/_common/SectionMessage/SectionMessage.jsx

```diff
@@ -3,38 +3,46 @@ import PropTypes from 'prop-types';
 import Message from '_common/Message';
 
 /**
- * Show an event-based message to the user
- * @todo Document examples
+ * Show a section/page-specific event-based message to the user
  * @example
- * // Blah blah…
- * <Sample jsx>
+ * // basic usage
+ * <SectionMessage type="warning">Uh oh.</SectionMessage>
+ * @see _common/Message
  */
 const SectionMessage = (props) => {
   const [isVisible, setIsVisible] = useState(true);
+  const autoManageVisible = props.canDismiss && props.isVisible === undefined;
+  const autoManageDismiss = props.canDismiss && props.onDismiss === undefined;
 
-  // Manage visibility
-  const onDismiss = useCallback(() => {
-    setIsVisible(!isVisible);
-  }, [isVisible]);
+  function onDismiss() {
+    if (autoManageVisible) {
+      setIsVisible(!isVisible);
+    }
+    if (!autoManageDismiss) {
+      props.onDismiss();
+    }
+  }
 
   // Override default props
-  const messageProps = {
+  let messageProps = {
     ...Message.defaultProps,
     ...props,
-    isVisible,
-    onDismiss,
+    onDismiss: onDismiss,
     scope: 'section',
   };
+  if (autoManageVisible) {
+    messageProps.isVisible = isVisible;
+  }
+  if (autoManageDismiss) {
+    messageProps.onDismiss = onDismiss;
+  }
 
   // Avoid manually syncing <Message>'s props
   // eslint-disable-next-line react/jsx-props-no-spreading
   return <Message {...messageProps} />;
 };
-SectionMessage.propTypes = {
-  ...Message.propTypes,
-  isVisible: PropTypes.bool,
-  onDismiss: PropTypes.func,
-};
-SectionMessage.defaultProps = Message.defaultProps;
+SectionMessage.propTypes = Message.propTypes;
+let { isVisible, onDismiss, ...defaultProps } = Message.defaultProps;
+SectionMessage.defaultProps = defaultProps;
```

Comment:
Why not just set it to `undefined` by default? (like in [FormField.jsx](https://github.com/TACC/Core-Portal/blob/5adcb06152790c35a61fd2f86724b87efe1facec/client/src/components/_common/Form/FormField.jsx#L166))
I think this is a bit more readable imo unless I'm missing something.
```js
SectionMessage.defaultProps = {
  ...Message.defaultProps,
  isVisible: undefined,
  onDismiss: undefined
};
```

Reply:
Because I overcomplicate things. Yes, that is more readable.

I now use your solution, because it does not create unused variables like my solution. Thanks.